### PR TITLE
[Calendar] add google accept decline meeting

### PR DESCRIPTION
**Pull request by Theodosia1005**
## Description

Add google accept and decline meeting

## Testing Steps

accept a meeting
decline a meeting

## Checklist
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the appropriate unit tests
- [ ] I have updated related documentation

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant
- [ ] A duplicate issue is filed to track future  work.

If you have updated responses or `.lu` files:
- [ ] All languages have been updated
- [ ] You have tested deployment with your new models

### CODE REVIEW

**Review by KayMKM — DISMISSED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskilltest/API/Fakes/MockGoogleClient/MockGoogleServiceClient.cs [hunk 1]
```diff
@@ -73,6 +73,18 @@ static MockGoogleServiceClient()
 
                 return mockDeleteRequest;
             });
+
+            mockEventsResource.Setup(events => events.Get(It.IsAny<string>(), It.IsAny<string>())).Returns((string calendarId, string eventId) =>
+            {
+                if (calendarId != "primary")
+                {
+                    throw new Exception("Calendar ID not support");
```
Comment (darrenj) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskilltest/API/Fakes/MockGoogleClient/MockGoogleServiceClient.cs]: supported

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskilltest/API/Fakes/MockGoogleClient/MockGoogleServiceClient.cs [hunk 2]
```diff
@@ -319,6 +331,72 @@ public new string Execute()
...
+                            Self = null,
+                        },
+                    };
+                }
+
+                public new Event Execute()
+                {
+                    if (CalendarId != "primary")
+                    {
+                        throw new Exception("Calendar ID not support");
```
Comment (darrenj) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskilltest/API/Fakes/MockGoogleClient/MockGoogleServiceClient.cs]: supported

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs [hunk 3]
```diff
@@ -266,5 +270,51 @@ private string DeleteEvent(string id)
                 throw GoogleClient.HandleGoogleAPIException(ex);
             }
         }
+
+        private void DeclineEvent(string id)
+        {
+            try
+            {
+                var request = _service.Events.Get("primary", id);
```
Comment (darrenj) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs]: move strings to a central file of const strings (or follow patterns used elsewhere) especially if used elsewhere

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs [hunk 4]
```diff
@@ -266,5 +270,51 @@ private string DeleteEvent(string id)
...
+                var gevent = ((IClientServiceRequest<Event>)request).Execute();
+                foreach (var attendee in gevent.Attendees)
+                {
+                    if (attendee.Self.HasValue && attendee.Self.Value)
+                    {
+                        attendee.ResponseStatus = "declined";
```
Comment (darrenj) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs]: Review this, should this be a ResponseStatus enumeration or similar?

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs [hunk 5]
```diff
@@ -266,5 +270,51 @@ private string DeleteEvent(string id)
...
+        {
+            try
+            {
+                var request = _service.Events.Get("primary", id);
+                var gevent = ((IClientServiceRequest<Event>)request).Execute();
+                foreach (var attendee in gevent.Attendees)
+                {
+                    if (attendee.Self.HasValue && attendee.Self.Value)
+                    {
+                        attendee.ResponseStatus = "accepted";
```
Comment (darrenj) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs]: see earlier comment regarding enumeration?

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/Models/EventModel.cs [hunk 6]
```diff
@@ -767,32 +777,42 @@ public EventStatus Status
...
-                        else
-                        {
-                            gmailEventData.Status = "tentative";
+                            if (attendee.Self.HasValue && attendee.Self.Value)
+                            {
+                                switch (value)
+                                {
+                                    case EventStatus.Accepted:
+                                        attendee.ResponseStatus = "accepted";
+                                        break;
```
Comment (darrenj) [inline solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/Models/EventModel.cs:807]: follow same enumeration approach as done above for microsoft?

**Review by darrenj — CHANGES_REQUESTED** [review]

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs (hunk 4 above)
Comment (Theodosia1005) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs]: change to const class.

File: solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs (hunk 3 above)
Comment (Theodosia1005) [reply solutions/Virtual-Assistant/src/csharp/skills/calendarskill/calendarskill/ServiceClients/GoogleAPI/GoogleCalendarAPI.cs]: change to const

**Review by darrenj — APPROVED** [review]

**Pull request merged**